OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.java

```diff
@@ -139,6 +172,104 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         }
     }
 
+    private void setupIAPLayout() {
+        binding.accessError.setVisibility(View.VISIBLE);
+        binding.accessError.setState(EdxCourseAccessErrorState.State.IS_UPGRADEABLE);
+        binding.toolbar.tabs.setVisibility(View.GONE);
+        setupToolbar(true);
+
+        iapViewModel = new ViewModelProvider(this).get(InAppPurchasesViewModel.class);
+        boolean isPurchaseEnabled = environment.getAppFeaturesPrefs().isIAPEnabled(environment.getLoginPrefs().isOddUserId());
+        if (isPurchaseEnabled) {
+            initIAPObservers();
+            // Shimmer container taking sometime to get ready and perform the animation, so
+            // by adding the some delay fixed that issue for lower-end devices, and for the
+            // proper animation.
+            binding.accessError.postDelayed(
+                    () -> iapViewModel.initializeProductPrice(courseData.getCourseSku()), 1500);
+        }
+        binding.accessError.setSecondaryButtonListener(onFindCourseClick());
+    }
+
+    private void initIAPObservers() {
+        iapViewModel.getProductPrice().observe(this, skuDetailsEvent -> {
```

Comment:
every observer is giving a warning to use `getViewLifecycleOwner()` instead `this`

Reply:
How about using EventObservers instead of manually checking?

Reply:
Same on other observers.

Reply:
> every observer is giving a warning to use getViewLifecycleOwner() instead this

`getViewLifecycleOwner()` and `this` both are the same at this point.

> How about using EventObservers instead of manually checking?

The `EventObservers` and `Event` are both kotlin classes, I didn't find a direct way to use these.

Reply:
updated. 👍